api/envoy/api/v2/route/route_components.proto

```diff
@@ -1051,6 +1051,10 @@ message RetryPolicy {
 
   // HTTP headers which must be present in the request for retries to be attempted.
   repeated HeaderMatcher retriable_request_headers = 10;
+
+  // Implementation specific configuration which depends on the implementation being instantiated.
+  // See the supported retry policy implementations for further documentation.
+  google.protobuf.Any typed_config = 11;
```

Comment:
AFAICT none of the above configuration is made available to the retry extension. Can you help clarify which aspects of the retry config is passed to the extension and what is ignored, etc.? I'm mainly trying to figure out if we have an API issue in which we are sticking the retry config in the wrong place, per previous discussion? Thank you.

/wait-any

Reply:
> AFAICT none of the above configuration is made available to the retry extension.

I agree that retry extension doesn't need to access the above configuration but the `retry_state_impl` requires the above configuration. If we use oneof, the `retry_state_impl` cannot get those configurations.

Let's consider the field `num_retries`. `retry_state_impl` needs to initialize the number of retries and it reads the value from `retry_policy`. If we use oneof, `retry_state_impl` cannot get the value of `num_retries`.

Reply:
OK I think per-offline convo, it's not clear to me as a user when I might want to use/implement an extension. Can you add more docs on what an extension might do and maybe an example extension also that is used in an integration test? That would be super helpful. Thank you!

/wait

Reply:
I think we'll want to keep it within the retry policy for now, potentially deprecating other fields from this proto once they are implemented as extensions

Reply:
I still think this needs more docs. Just because I'm dense: do all the above fields apply here? Or do some of them not apply if this is set? Can the docs clarify? Also please add a general overview about this into the arch overview docs for HTTP somewhere, and please also update https://www.envoyproxy.io/docs/envoy/latest/extending/extending

Reply:
Working on doc

Reply:
Please ref link to the new docs you added from the field docs.